javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll

```diff
@@ -32,6 +32,7 @@ module DataFlow {
   or TReflectiveCallNode(MethodCallExpr ce, string kind) {
        ce.getMethodName() = kind and (kind = "call" or kind = "apply")
      }
+  or TThisNode(StmtContainer f) { f.(Function).getThisBinder() = f or f instanceof TopLevel }
```

Comment:
We could do marginally better here by unifying the `this` expressions of inlined scripts:

```html
<script>this</script>
<script>this</script>
```

I do not think it is worth it in practice though.

Reply:
Agreed. It's a hassle to support that, and it still wouldn't be a good vehicle for tracking flow through globals, so let's not go down that road.

Reply:
> It's a hassle to support that

Is it really? Couldn't you just associate the `TThisNode` with the `File` instead of the `TopLevel` in cases like this?

Of course, the likely benefits seem marginal, so if you don't think it's worth it, that's fine.

Reply:
If we're happy with the type of `getThisBindingContainer` being something generic like `Locatable` it would indeed be straightforward.

A better way to do it IMO would be to add a `TGlobalObjectNode(File file)` case, so we could handle things like `window` and `global` at the same time.

Reply:
Let's leave it as-is for now.